orion/primitives/azure_anomaly_detector.py

```diff
@@ -0,0 +1,175 @@
+        index (ndarray):
+            N-dimensional array containing the input index sequences.
+        interval (int):
+            Integer denoting time span frequency of the data.
+        pad_size (int):
+            Length of the values from previous sequence.
+        subscription_key (str):
+            Resource key for authenticating your requests.
+        endpoint (str):
+            Resource endpoint for sending API requests.
+        granularity (str or Granularity):
+            Can only be one of yearly, monthly, weekly, daily, hourly or minutely.
+            Granularity is used for verify whether input series is valid. Possible values
+            include: 'yearly', 'monthly', 'weekly', 'daily', 'hourly', 'minutely'.
+        custom_interval (int):
+            Integer used to set non-standard time interval, for example, if the series
+            is 5 minutes, request can be set as {"granularity":"minutely", "custom_interval":5}.
+            If not given, `None` is used.
+        period (int):
+            Periodic value of a time series. If not given, `None` is used, and the API
+            will determine the period automatically.
+        max_anomaly_ratio (float):
+            Advanced model parameter, max anomaly ratio in a time series. If not given,
+            `None` is used.
+        sensitivity (int):
+            Advanced model parameter, between 0-99, the lower the value is, the larger
+            the margin value will be which means less anomalies will be accepted. If not given,
+            `None` is used.
+
+        Returns:
+            list:
+                Array containing start-index, end-index, score for each anomalous sequence.
+                Note that the API does not have an anomaly score, and so score is set to `None`.
+    """
+
+    def _convert_date(x):
+        return datetime.fromtimestamp(x).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+    SUBSCRIPTION_KEY = subscription_key
+    ANOMALY_DETECTOR_ENDPOINT = endpoint
+
+    client = AnomalyDetectorClient(
+        ANOMALY_DETECTOR_ENDPOINT, CognitiveServicesCredentials(SUBSCRIPTION_KEY))
+
+    min_pad = 0
+    result = list()
+
+    for x, idx in zip(X, index):
+        series = []
+        for i in range(len(x)):
+            idx_ = _convert_date(idx[i])
+            series.append(Point(timestamp=idx_, value=x[i]))
+
+        request = Request(
+            series=series, granularity=granularity, custom_interval=custom_interval, period=period,
+            max_anomaly_ratio=max_anomaly_ratio, sensitivity=sensitivity)
+
+        try:
+            response = client.entire_detect(request)
+            if response.is_anomaly:
+                anomalous = response.is_anomaly[min_pad:]
+                index_ = idx[min_pad:]
+                result.extend(index_[anomalous])
+
+        except Exception as e:
+            if isinstance(e, APIErrorException):
+                print('Error code: {}'.format(e.error.code),
```

Comment:
Can we use `logging` instead of a `print` statement?

```
import logging

LOGGER = logging.getLogger(__name__)
...
    except Exception:
        LOGGER.error('Error')
```

Notice that by using the `LOGGER.exception` level, the full traceback will be automatically printed in the logs, so the error code and message will probably be visible already without the need to put them explicitly in the logged error message.